Image processing - Get ready to run

ProfBits opened the pull request:
This has to main parts:
1. improvemetn for cut_mask_from_image 
2.  getting Image-Processing ready to optimize over night

### Fix exeption in cut_mask_from_image if mask and image mixed grayscale and rgb

During integration in the image-processing approche, the cut_mask_from_image raised exeptions. This was,
because sometimes the image and mask had mixed grayscale and rgb formats.

To fix this, first it is checked if the total numer of elements is larger or smaller in one of them. If the
mask has to few, it is grayscale and shuld be rgb so the one channel is copied 3 times. In the ohter cases
the mask has is rgb and the imrage only grayscale, so any one channel of the mask is good enough.

###  Multiple improvements are made.

- The genetic algorithem now only evaluates the new configurations, reusing the old ones. This is a ~2x speed up
- The genetic algorithem now supports custom intermediate results files
- The genetic algorithen now supports external seeding of the first popolation
- The genetic algorithem now supports prepreared in memeory images to speed up calculation
- Added saturation and contrast to the parametrs and pipeleine.
- Added house removeal at the beginning and end.

For the first training round the shadow removal steps will not be optimized and precomputed to save time.
HSV detection is currently disable to improve convergence speed.

The results of a first 30 generations evolution are in current_bet_01_07.txt stored and will be used as
additional seed paramers in future runs.

ProfBits committed:
Fix exeption in cut_mask_from_image if mask and image mixed grayscale and rgb

During integration in the image-processing approche, the cut_mask_from_image raised exeptions. This was,
because sometimes the image and mask had mixed grayscale and rgb formats.

To fix this, first it is checked if the total numer of elements is larger or smaller in one of them. If the
mask has to few, it is grayscale and shuld be rgb so the one channel is copied 3 times. In the ohter cases
the mask has is rgb and the imrage only grayscale, so any one channel of the mask is good enough.

ProfBits committed:
Improve formatting

ProfBits committed:
Improve image processing approche and get ready to run

Multiple improvements are made.

- The genetic algorithem now only evaluates the new configurations, reusing the old ones. This is a ~2x speed up

- The genetic algorithem now supports custom intermediate results files

- The genetic algorithen now supports external seeding of the first popolation

- The genetic algorithem now supports prepreared in memeory images to speed up calculation

- Added saturation and contrast to the parametrs and pipeleine.

- Added house removeal at the beginning and end.

For the first training round the shadow removal steps will not be optimized and precomputed to save time.
HSV detection is currently disable to improve convergence speed.

The results of a first 30 generations evolution are in current_bet_01_07.txt stored and will be used as
additional seed paramers in future runs.

ProfBits committed:
Enable loading of partial settings and fix bug

ProfBits committed:
Prepear for overnight run

- increase mutation chance for on averate 2 mutations
- remove cutting houses from lables
- add preprocessing outputs
- increase iteration count

joleitner commented on image-processing/core/configurator.py:
```diff
@@ -82,19 +88,55 @@ def __continues_parameter(limits: tuple[any, any], default: any, is_int: bool =
 
-def __run(image: str | cv2.typing.MatLike, output: Optional[str], parameters: dict[str, any]) -> cv2.typing.MatLike:
+def __run(
+        image: str | cv2.typing.MatLike | dict[str, np.ndarray],
```
cv2.typing.MatLike does unfortunately not work with our conda environment. Maybe we change it to np.ndarray so that it does not complicate things. Otherwise we should at least mention it in a README

joleitner reviewed (approved):
Looks good, think about the cv2.typing though

ProfBits committed:
Change typehints from cv2.MatLike to np.ndarray

The vm just cannot find the cv2.typing.MatLike type reference, tof fix this the type hints are changed to np.ndarray.

ProfBits committed:
Add temp/ to .gitignore

temp/ is used as a temporary working directory and therefor should not be contained in any commit.

ProfBits committed:
Add the current time to the running output.

ProfBits committed:
Fix mask_evaluation import on the VM

The VM could not import the mask_evaluation package in optimize.py. To fix this, the only way found to work was sys.path.append. Not nice however works for now.

Why are python imports such a pain.

ProfBits committed:
Remove house mask from true masks in preparation step

This is as requested by the clients, as everything that is marked as a house shall never be considered green area.

ProfBits committed:
Adjust parameters for optimization.

- Mutation rate changes to expect 1 parameter mutation per crossover
- increase generations for longer rungs
- remove initial population seeding to reduce bias
- change work dir to temp/

ProfBits committed:
Add 2 new results from vm overnight runs

ProfBits committed:
Merge branch 'master' into image-processing-include-new-modules

joleitner reviewed (approved):
LGTM and thank you for the ndarray change!

ProfBits committed:
Merge branch 'master' into image-processing-include-new-modules
